packages/react/src/components/Accordion/AccordionItem.tsx

```diff
@@ -131,24 +129,33 @@ function AccordionItem({
   const className = cx({
     [`${prefix}--accordion__item`]: true,
     [`${prefix}--accordion__item--active`]: isOpen,
-    [`${prefix}--accordion__item--${animation}`]: animation,
     [`${prefix}--accordion__item--disabled`]: disabled,
     [customClassName]: !!customClassName,
   });
 
   const Toggle = renderToggle || renderExpando; // remove renderExpando in next major release
 
-  if (open !== prevIsOpen) {
-    setAnimation(isOpen ? 'collapsing' : 'expanding');
-    setIsOpen(open);
-    setPrevIsOpen(open);
-  }
-
   // When the AccordionItem heading is clicked, toggle the open state of the
   // panel
   function onClick(event) {
+    const toggleElement = event.target;
+
+    let content = toggleElement.nextElementSibling;
+
+    // when toggle element is not the button
+    if (toggleElement.nodeName !== 'BUTTON') {
+      content = toggleElement.parentElement.nextElementSibling;
+    }
```

Comment:
Gaining access to the content node this way feels brittle. Instead, could a ref be placed on the `div` containing the `children`, and then that `contentRef.current.style` be used in the `maxBlockSize` logic below?